tree/ntuple/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -65,13 +65,142 @@ public:
    std::unique_ptr<ROOT::Internal::RPageSource> CreatePageSource() const;
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleProcessorOptionalPtr<T>
+\ingroup NTuple
+\brief The RNTupleProcessorOptionalPtr provides access to values from fields present in an RNTupleProcessor, with support
+and checks for missing values.
+*/
+// clang-format on
+template <typename T>
+class RNTupleProcessorOptionalPtr {
+   friend class RNTupleProcessor;
+
+private:
+   Internal::RNTupleProcessorEntry *fProcessorEntry;
+   Internal::RNTupleProcessorEntry::FieldIndex_t fFieldIndex;
+
+   RNTupleProcessorOptionalPtr(Internal::RNTupleProcessorEntry *processorEntry,
+                               Internal::RNTupleProcessorEntry::FieldIndex_t fieldIdx)
+      : fProcessorEntry(processorEntry), fFieldIndex(fieldIdx)
+   {
+   }
+
+public:
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Check if the pointer currently holds a valid value.
+   bool HasValue() const { return fProcessorEntry->IsValidField(fFieldIndex); }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get a shared pointer to the field value managed by the processor's entry.
+   ///
+   /// \return A `std::shared_ptr<T>` if the field is valid in the current entry, or a `nullptr` otherwise.
+   std::shared_ptr<T> GetPtr() const
+   {
+      if (fProcessorEntry->IsValidField(fFieldIndex))
+         return fProcessorEntry->GetPtr<T>(fFieldIndex);
+
+      return nullptr;
+   }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get a raw pointer to the field value managed by the processor's entry.
+   ///
+   /// \return A `T*` if the field is valid in the current entry, or a `nullptr` otherwise.
+   T *GetRawPtr() const { return GetPtr().get(); }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Bind the value to `valuePtr`.
+   ///
+   /// \param[in] valuePtr Pointer to bind the value to.
+   void BindRawPtr(T *valuePtr) { fProcessorEntry->BindRawPtr(fFieldIndex, valuePtr); }
```

Comment:
I'm trying to think about what use case this serves. `RNTupleProcessorOptionalPtr` is a public class and this is a public method. So I am inferring that the RNTupleProcessorOptionalPtr can be used as a detached object that binds its internal entry to a raw pointer? But then, the internal entry is internal, so only an RNTupleProcessor can set it, so calling `BindRawPtr` from the user side may result in accessing invalid memory. This is just free-flow of thoughts, maybe there is some chance of clarifying the intent here.

Reply:
You are right, but to also allow the processor to be used in core frameworks, during an earlier offline discussion we came to the conclusion that it has to be possible for it to read into user-provided pointers. It is true that this comes with the risk of reading invalid values and the responsibility to prevent this is moved to the user (which can still be achieved by only reading through the `RNTupleProcessorOptionalPtr`). This is an interface for advanced use only and will not be advertised. Thinking about it again we could consider moving this functionality to a free function in the Detail namespace, but then we also need to change again the option to pass a pointer to `RequestField`. I will add a clarifying comment for now.